### PR TITLE
Added client package

### PR DESCRIPTION
**Pull request by hlts2**
### Description:

I added vald and agent client package.
These are used by benchmark packages under hack and by the valdcli tool.
The interface is similar to the proto file.

### Environment:

- Golang Version: 1.13.5
- Docker Version: 19.03.5
- Kubernetes Version: 1.16.3
- NGT Version: 1.8.4

### Types of changes:

- [ ] Bug fix [type/bug]
- [x] New feature [type/feature]
- [ ] Add tests [type/test]
- [ ] Security related changes [type/security]
- [ ] Add documents [type/documentation]
- [ ] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Have you successfully ran tests with your changes locally?

### Checklist:

- [ ] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [ ] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [ ] I have added tests and benchmarks to cover my changes.
- [ ] I have ensure all new and existing tests passed.
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: internal/client/gateway/option.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
+package gateway
+
+import "github.com/vdaas/vald/internal/config"
+
+type Option func(*gatewayClient)
+
+var (
+	defaultOptions = []Option{
+		WithAddr("0.0.0.0"),
```
Comment (kmrmt) [inline internal/client/gateway/option.go]: same

File: internal/client/agent/option.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
+package agent
+
+import "github.com/vdaas/vald/internal/config"
+
+type Option func(*agentClient)
+
+var (
+	defaultOptions = []Option{
+		WithAddr("0.0.0.0"),
```
Comment (kmrmt) [inline internal/client/agent/option.go]: Client needs to connect server with port.

Comment (hlts2) [reply internal/client/agent/option.go]: oh:bow:
I will fix it
thank you for review.

File: internal/net/grpc/stream.go [hunk 3]
```diff
@@ -72,5 +72,37 @@ func BidirectionalStream(stream grpc.ServerStream,
 	}
 }
 
-// TODO implement BidirectionalStreamClient
-// func BidirectionalStreamClient(...) error {}
\ No newline at end of file
+func BidirectionalStreamClient(stream grpc.ClientStream,
+	concurrency int,
+	newData func() interface{},
```
[the remark this answers is not in the document]
Comment (hlts2) [reply internal/net/grpc/stream.go]: Thank you for your review.
I think so. thank you.
I will fix it soon.

[the remark this answers is not in the document]
Comment (hlts2) [reply internal/net/grpc/stream.go]: However, in terms of generating data, I thought newdata could be just as good as a `BidirectionalStream`.
Also, it is self-evident that data is generated when using the function `function () interface {}`. So I think I don't need to add it to the variable name. :thinking:

[the remark this answers is not in the document]
Comment (hlts2) [reply internal/net/grpc/stream.go]: kpango said `dataprovider` is better in English. 
So I fixed to `dataProvider`.
Sorry:bow:

**Pull request closed without merging**